src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -184,20 +187,7 @@ SHAMapStoreImp::SHAMapStoreImp(
 
     if (deleteInterval_)
     {
-        // Configuration that affects the behavior of online delete
-        get_if_exists(section, "delete_batch", deleteBatch_);
-        std::uint32_t temp;
-        if (get_if_exists(section, "back_off_milliseconds", temp) ||
-            // Included for backward compaibility with an undocumented setting
-            get_if_exists(section, "backOff", temp))
-        {
-            backOff_ = std::chrono::milliseconds{temp};
-        }
-        if (get_if_exists(section, "age_threshold_seconds", temp))
-            ageThreshold_ = std::chrono::seconds{temp};
-        if (get_if_exists(section, "recovery_wait_seconds", temp))
-            recoveryWaitTime_.emplace(std::chrono::seconds{temp});
-
+        assert(!reportingReadOnly_);
```

Comment:
Yes this should throw. This should actually throw in reporting mode, whether read only or not. Reporting mode does not do online delete and does not rotate anything. Setting `delete_interval_` will cause `DatabaseRotating` to be used instead of `DatabaseNodeImp`. Furthermore, throwing when encountering any config sections that pertain to online delete when running in reporting mode is better than accepting those config params and silently not deleting anything.

Since usually many reporting servers are sharing access to the same database, deletion should not occur in application code, as the servers would need to coordinate with each other (which can get hairy).

Reply:
Yes this should throw. This should actually throw in reporting mode, whether read only or not. Reporting mode does not do online delete and does not rotate anything. Setting `delete_interval_` will cause `DatabaseRotating` to be used instead of `DatabaseNodeImp`. Furthermore, throwing when encountering any config sections that pertain to online delete when running in reporting mode is better than accepting those config params and silently not deleting anything.

Since usually many reporting servers are sharing access to the same database, deletion should not occur in application code, as the servers would need to coordinate with each other (which can get hairy).